TAKAware/Screens/SettingsScreens/KMLOptions.swift

```diff
@@ -0,0 +1,113 @@
+    @State var isShowingFilePicker = false
+    @State var isShowingAlert = false
+    @State var alertText: String = ""
+    
+    @FetchRequest(sortDescriptors: [])
+    var kmlFiles: FetchedResults<KMLFile>
+    
+    var body: some View {
+        List {
+            Button {
+                isShowingFilePicker.toggle()
+            } label: {
+                HStack {
+                    Text("Import a KML")
+                    Spacer()
+                    Image(systemName: "square.and.arrow.up")
+                        .multilineTextAlignment(.trailing)
+                }
+                
+            }
+            .buttonStyle(.plain)
+            .contentShape(Rectangle())
+            .fileImporter(isPresented: $isShowingFilePicker, allowedContentTypes: [UTType(filenameExtension: "kml")!, UTType(filenameExtension: "kmz")!], allowsMultipleSelection: false, onCompletion: { results in
+                
+                switch results {
+                case .success(let fileurls):
+                    for fileurl in fileurls {
+                        if(fileurl.startAccessingSecurityScopedResource()) {
+                            TAKLogger.debug("Processing KML at \(String(describing: fileurl))")
+                            let kmlImporter = KMLImporter(archiveLocation: fileurl)
+                            Task {
+                                let didSucceed = await kmlImporter.process()
+                                if didSucceed {
+                                    alertText = "Data package processed successfully!"
+                                } else {
+                                    alertText = "Data package processing failed"
+                                }
+                            }
+                            fileurl.stopAccessingSecurityScopedResource()
+                            isShowingAlert = true
+                        } else {
+                            TAKLogger.error("Unable to securely access \(String(describing: fileurl))")
+                        }
+                    }
+                case .failure(let error):
+                    TAKLogger.debug(String(describing: error))
+                }
+                
+            })
+            Section(
+                header: Text("Imported KML Files"),
+                footer: Text("Swipe a file to manage")
+            ) {
+                if kmlFiles.isEmpty {
+                    Text("No KML Files Imported")
+                } else {
+                    ForEach(kmlFiles) { kmlFile in
+                        HStack {
+                            Image(systemName: "doc.text.fill")
+                            VStack(alignment: .leading) {
+                                Text(kmlFile.fileName ?? "Unknown Name")
+                                    .fontWeight(.bold)
+                            }
+                        }
+                        .swipeActions(allowsFullSwipe: false) {
```

Comment:
Can we implement this in the data package details list as well for KML/KMZ files?